core/templates/dev/head/pages/email_dashboard/EmailDashboard.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for oppia email dashboard page.
+ */
+
+oppia.controller('EmailDashboard', [
+  '$scope', '$http', function($scope, $http) {
+    var QUERY_DATA_URL = '/emaildashboarddatahandler';
+
+    // Store current cursor.
+    $scope.currentCursor = null;
```

Comment:
Oh, re-check status sounds good but in that case we will need entire new handler. Which will take input 'query_id' and then send latest info from query model. Since queries are listed from a list, entry of this query in the list will have to be updated (in both $scope.queries and $scope.recentQueries). I think this can be done.